app/uk/gov/hmrc/nisp/services/admin/FeatureFlagService.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nisp.services.admin
+
+import uk.gov.hmrc.nisp.config.ApplicationConfig
+import uk.gov.hmrc.nisp.models.admin.{FeatureFlag, FeatureFlagName}
+import play.api.cache.AsyncCacheApi
+import uk.gov.hmrc.nisp.repositories.admin.FeatureFlagRepository
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.duration.{Duration, FiniteDuration, SECONDS => Seconds}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class FeatureFlagService @Inject()(
+                                    appConfig: ApplicationConfig,
+                                    featureFlagRepository: FeatureFlagRepository,
+                                    cache: AsyncCacheApi
+                                  )(implicit
+                                    ec: ExecutionContext
+                                  ) {
+  val cacheValidFor: FiniteDuration =
+    Duration(appConfig.ehCacheTtlInSeconds, Seconds)
+  private val allFeatureFlagsCacheKey = "*$*$allFeatureFlags*$*$"
+
+  def set(flagName: FeatureFlagName, enabled: Boolean): Future[Boolean] =
+    for {
+      _ <- cache.remove(flagName.toString)
+      _ <- cache.remove(allFeatureFlagsCacheKey)
+      result <- featureFlagRepository.setFeatureFlag(flagName, enabled)
+      //blocking thread to let time to other containers to update their cache
+      _ <- Future.successful(Thread.sleep(appConfig.ehCacheTtlInSeconds * 1000))
+    } yield result
+
+  def get(flagName: FeatureFlagName): Future[FeatureFlag] =
+    cache.getOrElseUpdate(flagName.toString, cacheValidFor) {
+      featureFlagRepository
+        .getFeatureFlag(flagName)
+        .map(_.getOrElse(FeatureFlag(flagName, false)))
+    }
+
+  def getAll: Future[List[FeatureFlag]] =
```

Comment:
I fixed a bug where if for some reason a toggle is deleted in the code. The mongo entry cannot be mapped back into the object and causing an internal server error.

See below. If the toggle does not exist anymore it gets deleted
https://github.com/hmrc/pertax-frontend/blob/f4952d7789d52b25fb3e52a3bc0207a4d0865c66/app/services/admin/FeatureFlagService.scala#L57